Add support for network interface user friendly name

jukkar opened the pull request:
If `CONFIG_NET_INTERFACE_NAME` is enabled (default is y), then the system will automatically set a user friendly name to the network interface like eth0, wlan0 etc. Application can change the interface name if needed.

This is useful so that applications do not need to try to use the device name for figuring out the desired network interface.

jukkar committed:
net: Add name support to network interface

If CONFIG_NET_INTERFACE_NAME is enabled (default is y), then
system will automatically set a user friendly name to the network
interface like eth0, wlan0 etc.
Application can change the interface name if needed.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
tests: net: iface: Add unit test for interface name support

Check that we can set and get the network interface name.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: shell: Print user friendly network interface name

If network interface name support is enabled, print the name
when showing network interface data.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

rlubos commented on include/zephyr/net/net_if.h:
```diff
@@ -2575,6 +2583,45 @@ bool net_if_is_wifi(struct net_if *iface);
+ * @details If interface name support is not enabled, empty string is returned.
+ *
+ * @param iface Pointer to network interface
+ * @param buf User supplied buffer
```
Missing `len` documentation.

rlubos commented on subsys/net/ip/net_if.c:
```diff
@@ -4732,6 +4733,147 @@ struct net_if *net_if_get_first_wifi(void)
+	}
+
+	name_len = strlen(net_if_get_config(iface)->name);
+	if (name_len > len) {
```
This doesn't seem to take into account NULL termination (`name_len` does not count it in).

rlubos commented on subsys/net/ip/net_if.c:
```diff
@@ -4732,6 +4733,147 @@ struct net_if *net_if_get_first_wifi(void)
+		return -ERANGE;
+	}
+
+	strncpy(buf, net_if_get_config(iface)->name, name_len);
```
Should this use `len` instead (the actual length of the provided buffer)?

jukkar replied:
It should not matter here as we have made sure in the check few lines before, that `name_len` fits into `len` size buffer.

rlubos commented on subsys/net/ip/net_if.c:
```diff
@@ -4732,6 +4733,147 @@ struct net_if *net_if_get_first_wifi(void)
+
+	memset(net_if_get_config(iface)->name, 0,
+	       CONFIG_NET_INTERFACE_NAME_LEN + 1);
+	strncpy(net_if_get_config(iface)->name, buf, name_len);
```
Same here, `name_len` does not count NULL terminator, so it'd be omitted (although in this particular case memset above guarantees the NULL terminator will be in the buffer anyway). I think `sizeof(net_if_get_config(iface)->name)` would be more appropriate.

rlubos commented on subsys/net/ip/net_if.c:
```diff
@@ -4732,6 +4733,147 @@ struct net_if *net_if_get_first_wifi(void)
+#endif /* CONFIG_NET_L2_PPP */
+
+#if defined(CONFIG_NET_L2_WIFI)
+	if (net_if_l2(iface) == &NET_L2_GET_NAME(WIFI)) {
```
We don't have a proper "WIFI" L2 yet, so this won't work. Wi-Fi interfaces are exposed as "ETHERNET" with a flag set on an interface that indicate the interface is WiFi, there is a dedicated function to check that (`net_if_is_wifi()`).

rlubos commented on subsys/net/ip/net_if.c:
```diff
@@ -4732,6 +4733,147 @@ struct net_if *net_if_get_first_wifi(void)
+	}
+
+	memset(net_if_get_config(iface)->name, 0,
+	       CONFIG_NET_INTERFACE_NAME_LEN + 1);
```
Maybe use `sizeof(net_if_get_config(iface)->name)` instead of `CONFIG_NET_INTERFACE_NAME_LEN + 1` so there's no need to carry +1 here and there?

jukkar replied:
Unfortunately `sizeof()` is done at compile time so the parameter cannot be a function call.

jukkar replied:
But I think we could access the config directly instead of using a function, I will try that.

rlubos replied:
True, I overlooked this involves a function call within `sizeof()`. If this doesn't fly, then no big deal, it isn't a major issue.

rlubos reviewed (commented):
Looks good, some minor comments. I think it'd make sense to update the `SO_BINDTODEVICE` implementation as well, for now it relies on the device name, but it should use the interface name instead I believe.

jukkar committed:
net: socket: Change SO_BINDTODEVICE to use interface name

Make sure we use the network interface name (if configured)
instead of device name when binding to certain network
interface.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

rlubos reviewed (dismissed):
LGTM :+1:

krish2718 commented on subsys/net/ip/Kconfig:
```diff
@@ -859,6 +859,21 @@ config NET_DEFAULT_IF_WIFI
+
+config NET_INTERFACE_NAME_LEN
+	int "Network interface max name length"
+	default 8
```
Should we sync with Linux here 1..15 for portability?

jukkar replied:
Sure, we can increase the max size.

jukkar commented:
Increased the max name length to 15 chars.
